fpdf/svg.py

```diff
@@ -566,293 +548,74 @@ def convert_transforms(tfstr):
-        # technically, it would also be possible to chain on from a quadratic Bézier,
-        # since we can convert those to cubic curves and then retrieve the appropriate
-        # control point. However, the SVG specification states in
-        # https://www.w3.org/TR/SVG/paths.html#PathDataCubicBezierCommands
-        # "if the previous command was not an C, c, S or s, assume the first control
-        # point is coincident with the current point."
-        if isinstance(last_item, BezierCurve):
-            c1 = (2 * last_item.end) - last_item.c2
-        else:
-            c1 = last_item.end_point
-
-        return BezierCurve(c1, self.c2, self.end).render(
-            path_gsds, style, last_item, initial_point
-        )
-
-    def render_debug(
-        self, path_gsds, style, last_item, initial_point, debug_stream, pfx
-    ):
-        # pylint: disable=unused-argument
-        rendered, resolved, initial_point = self.render(
-            path_gsds, style, last_item, initial_point
-        )
-        debug_stream.write(f"{self} resolved to {resolved}\n")
-
-        return rendered, resolved, initial_point
-
-
-@force_nodocument
-class SVGRelativeSmoothCubicCurve(NamedTuple):
-    """SVG chained relative cubic Bézier curve path element."""
-
-    c2: Point
-    end: Point
-
-    @classmethod
-    def from_path_points(cls, path, c2x, c2y, ex, ey):
-        return path.add_path_element(cls(c2=Point(x=c2x, y=c2y), end=Point(x=ex, y=ey)))
-
-    def render(self, path_gsds, style, last_item, initial_point):
-        last_point = last_item.end_point
-
-        if isinstance(last_item, BezierCurve):
-            c1 = (2 * last_item.end) - last_item.c2
-        else:
-            c1 = last_point
-
-        c2 = last_point + self.c2
-        end = last_point + self.end
-
-        return BezierCurve(c1, c2, end).render(
-            path_gsds, style, last_item, initial_point
-        )
-
-    def render_debug(
-        self, path_gsds, style, last_item, initial_point, debug_stream, pfx
-    ):
-        # pylint: disable=unused-argument
-        rendered, resolved, initial_point = self.render(
-            path_gsds, style, last_item, initial_point
-        )
-        debug_stream.write(f"{self} resolved to {resolved}\n")
-
-        return rendered, resolved, initial_point
-
-
-@force_nodocument
-class SVGSmoothQuadraticCurve(NamedTuple):
-    """SVG chained quadratic Bézier curve path element."""
-
-    end: Point
-
-    @classmethod
-    def from_path_points(cls, path, ex, ey):
-        return path.add_path_element(cls(end=Point(x=ex, y=ey)))
-
-    def render(self, path_gsds, style, last_item, initial_point):
-        if isinstance(last_item, QuadraticBezierCurve):
-            ctrl = (2 * last_item.end) - last_item.ctrl
-        else:
-            ctrl = last_item.end_point
-
-        return QuadraticBezierCurve(ctrl, self.end).render(
-            path_gsds, style, last_item, initial_point
+class PathPen:
```

Comment:
Why not inherinting from `fontTools.pens.basePen.AbstractPen`?
_cf._ https://fonttools.readthedocs.io/en/latest/pens/basePen.html?fontTools.pens.basePen.AbstractPen#fontTools.pens.basePen.AbstractPen

If there is a reason not to inherit from this class, could you add a mention and a link to the pen protocol in this class docstring please? _cf._ https://fonttools.readthedocs.io/en/latest/pens/basePen.html

Reply:
I didn't even see that one...

I guess it's mostly a formality, but definitively good form to use the appropriate ABS.